File: matrixElementsSum/matrixElementsSum.java
import java.io.*;
import java.util.*;

class matrixElementsSum{

public static int getElementsSum(int[][] matrix) {
    ArrayList<Integer> exclusion = new ArrayList<Integer>();
    int total = 0;
    for (int i = 0; i < matrix.length; i++){
        for (int j = 0; j < matrix[i].length; j++){
            if (!(exclusion.contains(j))){
                total += matrix[i][j];
                if (matrix[i][j] == 0){
                    exclusion.add((Integer)j);
                }
            }
        }
    }
    return total;
}

public static void main(String[] args){
    int[][] test = {{1,0,1},{1,5,3},{0,4,0}};
    System.out.println(getElementsSum(test));
    }



}

File: matrixElementsSum/matrixElementsSum.ts
function matrixElementsSum(matrix: any[][]): number {
    let excluded: number[] = []
    let total: number = 0;
    for (let i = 0; i < matrix.length; i++){
        for (let j = 0; j < matrix[i].length; j++){
            if (matrix[i][j] === 0){
                excluded.push(j);
            }
            else if (excluded.indexOf(j) === -1){
                total = total + matrix[i][j];
            }
        }
    }
return total;
}

console.log(matrixElementsSum(
    [[0, 1, 1, 2],
    [0, 5, 0, 0],
    [2, 0, 3, 3]]
));

